apps/server/src/shared/repo/contextexternaltool/context-external-tool.repo.ts

```diff
@@ -56,10 +59,12 @@ export class ContextExternalToolRepo extends BaseDORepo<
 	private buildScope(query: ContextExternalToolQuery): ContextExternalToolScope {
 		const scope: ContextExternalToolScope = new ContextExternalToolScope();
 
-		if (query.contextId && query.schoolToolId) {
-			scope.byContextIdAndSchoolToolId(query.contextId, query.schoolToolId);
-		} else if (query.schoolToolId && !query.contextId) {
-			scope.bySchoolToolId(query.schoolToolId);
+		if (query.id) {
+			scope.byId(query.id);
+		} else if (query.contextId && query.schoolToolRef?.schoolToolId) {
+			scope.byContextIdAndSchoolToolId(query.contextId, query.schoolToolRef.schoolToolId);
+		} else if (query.schoolToolRef?.schoolToolId && !query.contextId) {
+			scope.bySchoolToolId(query.schoolToolRef.schoolToolId);
 		}
```

Comment:
This can be simplified:
Case 2 and 3 require both that query.schoolToolRef?.schoolToolId exists. The only difference is if query.contextId is defined.


Further the scope.byXXX methods in these cases only differenciate between setting the queryId or not. This can be merged into one method imo. we can talk about this